server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -274,16 +297,15 @@ private Chromosome getNextChromosome(final List<Chromosome> chromosomes, final C
         return null;
     }
 
-    private Chromosome loadChrById(final Long referenceId, final Long chromosomeId) {
+    private Chromosome fetchChromosomeById(final Reference reference, final Long chromosomeId) {
         if (chromosomeId == null) {
-            return getFirstChromosomeFromGenome(referenceId);
-        } else {
-            return referenceGenomeManager.loadChromosome(chromosomeId);
+            return reference.getChromosomes().get(0);
```

Comment:
actually this two branches can be joined with filter condition (`chromosomeId == null || chromosomeId.equals(chr.getId())`)

Reply:
Thank you. It's actually reasonable.